src/common.jl

```diff
@@ -47,3 +47,20 @@ end
 
 # Tracker's implementation of ldiv isn't good. We'll use Zygote's instead.
 zygote_ldiv(A::AbstractMatrix, B::AbstractVecOrMat) = A \ B
+
+function randnsimilar(rng, x::AbstractArray, dims...)
+    randn!(rng, similar(x, dims...))
+end
+
+function randnsimilar(rng, x::CuArray, dims...)
+    adapt(CuArray, randn(rng, eltype(x), dims...))
+end
```

Comment:
It seems we could just use
```suggestion
function adapt_randn(rng, x::AbstractArray, dims...)
    adapt(typeof(x), randn(rng, eltype(x), dims...)
end
```
without ever depending on CUDA?

Reply:
ohhhh, very nice! I'll try that!